apps/test/unit/javalab/JavalabEditorTest.js

```diff
@@ -239,6 +239,64 @@ describe('Java Lab Editor Test', () => {
           'file-1': 'Class2.java'
         });
       });
+
+      it('displays error message if file name is blank', () => {
+        const editor = createWrapper();
+        const javalabEditor = editor.find('JavalabEditor').instance();
+        store.dispatch(
+          setAllSources({
+            'Class1.java': {text: '', isVisible: true, isValidation: false}
+          })
+        );
+
+        javalabEditor.setState({
+          showMenu: false,
+          contextTarget: null,
+          editTabKey: 'file-0',
+          editTabFilename: 'Class1.java',
+          openDialog: 'renameFile',
+          orderedTabKeys: ['file-0'],
+          fileMetadata: {
+            'file-0': 'Class1.java'
+          }
+        });
+        // We are trying to rename Class1.java -> ''
+        javalabEditor.onRenameFile('');
+        expect(store.getState().javalab.renameFileError).to.not.be.null;
```

Comment:
Might be worth checking some of these conditions prior to your rename to confirm you're getting the expected state change?

Reply:
Good call, will add